src/http-client.ts

```diff
@@ -0,0 +1,78 @@
+import request from 'superagent';
+
+import reqlogger from './request-logger';
+import { RetryPromise, retryPromise } from './utils';
+
+export type RequestApiOptions = {
+  method?: string;
+  data?: any;
+  host: string;
+  version?: string;
+  contentType?: string;
+  authToken?: string;
+  logger?: any
+};
+
+export class HttpClient {
+  private _apiRequestsMap = new Map<string, RetryPromise<any>>();
```

Comment:
This map holds all promises that are in a retry state. Once a promise resolves/rejects, it is removed from the map. This is here so any pending retries can be stopped later.